### PR TITLE
MULE-17812: Refresh Dance not executed after TokenExpiredException

### PR DESCRIPTION
**Pull request by SebaElizalde**

### CODE REVIEW

**Comment by SebaElizalde** [comment]
--validate

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OAuthOperationMessageProcessor.java [hunk 1]
```diff
@@ -77,7 +76,7 @@ public OAuthOperationMessageProcessor(ExtensionModel extensionModel,
   @Override
   protected Mono<CoreEvent> doProcess(CoreEvent event, ExecutionContextAdapter<OperationModel> operationContext) {
     return super.doProcess(event, operationContext)
-        .onErrorResume(AccessTokenExpiredException.class, e -> {
+        .onErrorResume(Exception.class, e -> {
```
Comment (ndinu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OAuthOperationMessageProcessor.java]: I would add a test for this fix, where the AccessTokenExpiredException was the cause of another exception

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java [hunk 2]
```diff
@@ -503,6 +504,10 @@ public void doDispose() {
   }
 
   protected ExecutionMediator createExecutionMediator() {
+    if (executionMediatorFactory != null) {
+      return executionMediatorFactory.get();
+    }
+
     return new DefaultExecutionMediator(extensionModel, componentModel, connectionManager, muleContext.getErrorTypeRepository());
```
Comment (ndinu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java]: What if you always use a supplier, but when creating the instance, you create the default supplier?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java [hunk 3]
```diff
@@ -594,4 +599,8 @@ private boolean isInterceptedComponent(ComponentLocation location, InternalEvent
     }
     return false;
   }
+
+  public void setExecutionMediatorFactory(Supplier<ExecutionMediator> executionMediatorFactory) {
+    this.executionMediatorFactory = executionMediatorFactory;
```
Comment (ndinu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java]: Maybe move this to the constructor? this way you can make it final.

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java [hunk 4]
```diff
@@ -152,6 +152,7 @@
   protected ComponentExecutor componentExecutor;
   protected ReturnDelegate returnDelegate;
   protected PolicyManager policyManager;
+  private Supplier<ExecutionMediator> executionMediatorFactory;
```
Comment (ndinu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java]: I would call it executionMediatorSupplier

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java [hunk 5]
```diff
@@ -542,6 +542,10 @@ private void setCacheIdGenerator() {
 
   protected abstract ParameterValueResolver getParameterValueResolver();
 
+  public ConnectionManagerAdapter getConnectionManager() {
```
Comment (ndinu) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java]: Maybe we  should find a way not to expose this?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java [hunk 6]
```diff
@@ -503,6 +504,10 @@ public void doDispose() {
   }
 
   protected ExecutionMediator createExecutionMediator() {
+    if (executionMediatorFactory != null) {
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java]: remove this if. The factory should default to a supplier with the default behavior.

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OAuthOperationMessageProcessor.java (hunk 1 above)
Comment (SebaElizalde) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OAuthOperationMessageProcessor.java]: Added in the integration tests.

**Pull request merged**